File: backend/app/models/booking.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Booking(Base):
    __tablename__ = "bookings"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), index=True
    )
    listing_id: Mapped[int] = mapped_column(
        ForeignKey("listings.id", ondelete="CASCADE"), index=True
    )
    check_in: Mapped[date] = mapped_column(Date)
    check_out: Mapped[date] = mapped_column(Date)
    guests: Mapped[int]
    status: Mapped[str] = mapped_column(String(20), default="held")  # held|confirmed|cancelled
    total_price: Mapped[float]
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

File: frontend/src/types/booking.ts
import type { Destination } from './destination'
import type { Listing } from './listing'

export type BookingStatus = 'held' | 'confirmed' | 'cancelled'

export interface Booking {
  id: number
  check_in: string
  check_out: string
  guests: number
  status: BookingStatus
  total_price: number
  created_at: string
}

export interface BookingDetail extends Booking {
  listing: Listing
  destination: Destination
}
